database/config.go

```diff
@@ -345,6 +348,17 @@ func (t *indexStore) Get(indexName string) (*IndexConfig, error) {
 	return &idxopts, nil
 }
 
+func (t *indexStore) Update(cfg IndexConfig) error {
```

Comment:
I think `Replace` would be a better name because we are literally replacing the IndexConfig by another one. Also, it's consistent with TableInfoStore#Replace